include/nuttx/mutex.h

```diff
@@ -89,7 +89,14 @@ extern "C"
 
 static inline int nxmutex_init(FAR mutex_t *mutex)
 {
-  return nxsem_init(mutex, 0, 1);
+  int ret = _SEM_INIT(mutex, 0, 1);
+
+  if (ret < 0)
+    {
+      return _SEM_ERRNO(ret);
```

Comment:
```suggestion
      return -(_SEM_ERRNO(ret));
```
I mean that we need to define what is the return value of `nxmutex_init` and others. Is it `errno` value or `-errno` value